pyuvdata/uvflag/uvflag.py

```diff
@@ -1099,40 +1339,56 @@ def to_metric(self, convert_wgts=False, run_check=True,
             converting to metric mode.
 
         """
-        if self.mode == 'metric':
+        if self.mode == "metric":
             return
-        elif self.mode == 'flag':
+        elif self.mode == "flag":
             self.metric_array = self.flag_array.astype(np.float)
             self._set_mode_metric()
 
             if convert_wgts:
                 self.weights_array = np.ones_like(self.weights_array)
-                if self.type == 'waterfall':
+                if self.type == "waterfall":
                     for i, pol in enumerate(self.polarization_array):
-                        self.weights_array[:, :, i] *= ~and_rows_cols(self.flag_array[:, :, i])
-                elif self.type == 'baseline':
+                        self.weights_array[:, :, i] *= ~and_rows_cols(
```

Comment:
I do believe the `|=` operator here is better than `*=`.

Reply:
I tried that but got this error:
` TypeError: ufunc 'bitwise_or' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`

Reply:
Ah, yeah you're right, `weights_array` is not boolean. :man_facepalming: